### DISCUSSION TITLE
PrimeReact DataTable paginator is not working properly.

### DISCUSSION THREAD

**Discussion by geeky01adarsh** [primefaces/community#1124] [PrimeReact Templates] [answered]
When I tried using the props passed through the Datatable paginator it didn't work. Also, we don't have the onPageChange with the DataTable.

Framework:- Next
Code:- 
```js
<DataTable paginator rows={5} totalRecords={100} value={data} stripedRows />
```
Here `data` had 5 elements and the total no. of pages shown was 1 only, while it must be 20.

I wanted to make paginated API calls, and on-page changes rather than getting all the data simultaneously.

**Comment by buraksaglam089** [comment]
Hello , 

To make paginated API calls using PrimeReact's DataTable component, you can utilize the onPage prop. This prop allows you to define a callback function that gets triggered when the pagination state changes, enabling you to update your data or make API calls accordingly

Regards.

**Reply by geeky01adarsh** [reply]
The `totalRecords` is not working, i.e., if my data have 5 elements only then, I am not getting the option for the second page.

The main issue for me is I can't utilize `totalRecords`.

**Comment by geeky01adarsh** [comment, marked as answer]
It works after adding `lazy` keyword.

```js
<DataTable paginator rows={5} totalRecords={100} lazy value={data} stripedRows />
```

**Reply by MarielaRH** [reply]
I've been fighting for a long time with that, thanks!

**Reply by bmbravo** [reply]
Just spent 3 hours looking for this solution, thanks 😆

**Reply by aakashjoshi1386** [reply]
Yes this hack is working with respect to paging API, I've spent many hours. But this helped and save it. Thanks for posting!

**Reply by innotexak** [reply]
A life saver indeed. Thank you
